tests/aws/services/apigateway/test_apigateway_lambda_cfn.py

```diff
@@ -1,80 +1,304 @@
+            auth_type=awslambda.FunctionUrlAuthType.NONE
+        ).url
 
-        api.add_gateway_response(
-            "default-5xx-response",
-            type=apigateway.ResponseType.DEFAULT_5_XX,
-            response_headers={
-                "Access-Control-Allow-Origin": "'*'",
+        role = iam.Role(
+            stack,
+            "TargetRole",
+            assumed_by=iam.ServicePrincipal("events.amazonaws.com"),
+            inline_policies={
+                "ApiDestinationPolicy": iam.PolicyDocument(
+                    statements=[
+                        iam.PolicyStatement(
+                            actions=["events:InvokeApiDestination"], resources=["*"]
+                        )
+                    ]
+                )
             },
         )
 
+        # Outputs
+        cdk.CfnOutput(stack, "BackendFunctionName", value=backend.function_name)
+        cdk.CfnOutput(stack, "OAuthFunctionName", value=oauth_token_lambda.function_name)
         cdk.CfnOutput(stack, "ApiId", value=api.rest_api_id)
+        cdk.CfnOutput(stack, "RoleArn", value=role.role_arn)
+        cdk.CfnOutput(stack, "OAuthTokenUrl", value=oauth_token_url)
 
+        infra.add_cdk_stack(stack)
         with infra.provisioner() as prov:
+            time.sleep(30)
             yield prov
 
     @markers.aws.validated
-    @markers.snapshot.skip_snapshot_verify(
-        paths=[
-            "$..restapiEndpointC67DEFEA",
-        ]
-    )
-    def test_scenario_validate_infra(self, aws_client, infrastructure, snapshot):
+    def test_api_destination_complex_integration(self, aws_client, infrastructure, snapshot, auth):
         snapshot.add_transformer(snapshot.transform.key_value("ApiId"))
-        outputs = infrastructure.get_stack_outputs(stack_name="ApiGatewayStack")
+        snapshot.add_transformer(snapshot.transform.key_value("RoleArnZ"))
+
+        # Retrieve outputs
+        outputs = infrastructure.get_stack_outputs(stack_name="ApiDestCfnTestStack")
+        backend_function_name = outputs["BackendFunctionName"]
+        oauth_function_name = outputs["OAuthFunctionName"]
         api_id = outputs["ApiId"]
-        apis = aws_client.apigateway.get_rest_api(restApiId=api_id)
-        assert apis["id"] == api_id
-
-        resources = infrastructure.get_stack_outputs(stack_name="ApiGatewayStack")
-        snapshot.match("resources", resources)
-
-        # makes sure we have a physical resource
-        resources = aws_client.cloudformation.describe_stack_resources(StackName="ApiGatewayStack")[
-            "StackResources"
-        ]
-        for r in resources:
-            if r["ResourceType"] == "AWS::ApiGateway::GatewayResponse":
-                assert r["PhysicalResourceId"]
+        role_arn = outputs["RoleArn"]
+        oauth_token_url = outputs["OAuthTokenUrl"]
+
+        region = aws_client.events.meta.region_name
+        invocation_endpoint = f"https://{api_id}.execute-api.{region}.amazonaws.com/prod/test"
```

Comment:
question: are we automatically transforming the endpoint in `events`? Because this won't resolve in the community edition of LocalStack, and the request will be sent to AWS.